### ISSUE TITLE
Spaces in local NCBI BLAST+ -outfmt argument (picking columns)

### ISSUE THREAD

**Issue by Hua-CM** [biopython/biopython#2055]
### Setup

I am reporting a problem with Biopython version, Python version, and operating
system as follows:

3.7.0 (default, Jun 28 2018, 13:15:42) 
[GCC 7.2.0]
CPython
Linux-4.18.0-17-generic-x86_64-with-debian-buster-sid
Biopython version 1.73

### Expected behaviour
I'd like to use BLAST in local envirnment, so I use the following codes:
from Bio.Blast.Applications import NcbiblastpCommandline
blastp_cline=NcbiblastpCommandline(query="a.fasta",
                                         db="../db/blast_db/database",
                                         outfmt="7 qacc sacc length pident evalue qcovs bitscore",
                                         num_threads=4,
                                         evalue=0.001,
                                         out="reverse_blast_result.txt")
blastp_cline()

### Actual behaviour
However ,my codes cause an error and I use print(blastp_cline) find the crux of problem.
The print(blastp_cline) result was as following:
blastp -out reverse_blast_result.txt -outfmt 7 qacc sacc length pident evalue qcovs bitscore -query a.fasta -db ../db/blast_db/database -evalue 0.001 -num_threads 4

The right command in shell should be:
blastp -out reverse_blast_result.txt -outfmt "7 qacc sacc length pident evalue qcovs bitscore" -query a.fasta -db ../db/blast_db/database -evalue 0.001 -num_threads 4

Could you tell me how to fix this problem?

**Comment by peterjc** [comment]
As a quick work around, try putting the double quotes in your string (by wrapping it in single quotes):

```
...
outfmt='"7 qacc sacc length pident evalue qcovs bitscore"',
...
```

or equivalently, using slash escaping:

```
...
outfmt="\"7 qacc sacc length pident evalue qcovs bitscore\"",
...
```

The wrapper needs updating to mention the possibility of listing columns, and special casing the space accordingly - much like with a filename. In fact, including ``filename=True,`` here would probably work but would need a command about the real intension:

https://github.com/biopython/biopython/blob/biopython-173/Bio/Blast/Applications.py#L60

Do you fancy working on this?

**Comment by chris-rands** [comment]
`filename=True` is used elsewhere for non-filenames to escape them, e.g.

https://github.com/biopython/biopython/blob/109604b3a6f43c5a941746381069b7d2ede3b7e9/Bio/Phylo/Applications/_Phyml.py#L136

https://github.com/biopython/biopython/blob/109604b3a6f43c5a941746381069b7d2ede3b7e9/Bio/Align/Applications/_Mafft.py#L284

So adding `filename=True` might be an appropriate fix?

Or else perhaps rename the argument to `escape=True` and re-name `_escape_filename` to `_escape_string` etc.?

https://github.com/biopython/biopython/blob/biopython-173/Bio/Application/__init__.py#L727

**Comment by peterjc** [comment]
Good ideas - I thought that might be the case, but didn't look for examples.

Changing the argument name would be a possible API break for anyone defining their own command line wrappers using this framework, so best avoided.

However, clarifying the broader usage in the docstrings, and renaming the internal function from ``_escape_filename`` to something like ``_quote_spaces`` would be sensible.

**Comment by chris-rands** [comment]
ok @peterjc i could do this one? should we re-name `_escape_filename` to `_quote_spaces`? or add a `BiopythonDeprecationWarning` to `_escape_filename` and duplicate the code into a new `_quote_spaces` function?

Are people using the internal `_escape_filename` function? e.g.
https://github.com/search?p=1&q=%22from+Bio.Application+import+_escape_filename%22&type=Code

**Comment by peterjc** [comment]
@chris-rands yes please. Skimming that search, most usage was in copies or forks of Biopython code, so low risk to change - but at least one real world usage in https://github.com/fsimkovic/mxkit though (first hit) so let's not change the public API right now.

**Comment by peterjc** [comment]
This should be fixed with #2071 and will be part of Biopython 1.74 - thank you @chris-rands for the fix, and @Hua-CM for the clear bug report.
